### PR TITLE
Fix Prop Types Error #14837

### PR DESCRIPTION
**Pull request by papuruth**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description
As per issue no. #14837 I have fixed prop types errors for Button, ButtonGroup, Modal, and articlePage components.

## Related Tickets & Documents
#14837 

## QA Instructions, Screenshots, Recordings
I don't believe any new tests would be required for these changes.

## Added/updated tests?
- [x] No, as because it doesn't seem legit to write UT for these changes.

## [optional] What gif best describes this PR or how it makes you feel?

Two on Behance

### CODE REVIEW

**Review by aitchiss — CHANGES_REQUESTED** [review]
Thanks for taking this on! It's looking great, but I just have a couple of small change requests - let me know if I can assist at all!

File: app/javascript/crayons/ButtonGroup/ButtonGroup.jsx [hunk 1]
```diff
@@ -10,5 +10,7 @@ export const ButtonGroup = ({ children }) => (
 ButtonGroup.displayName = 'ButtonGroup';
 
 ButtonGroup.propTypes = {
-  children: defaultChildrenPropTypes.isRequired,
+  children: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
+  ).isRequired,
```
[the remark this answers is not in the document]
Comment (papuruth) [reply app/javascript/crayons/ButtonGroup/ButtonGroup.jsx:15]: Exactly we should pass null @aitchiss

File: app/javascript/common-prop-types/default-children-prop-types.js [hunk 2]
```diff
@@ -4,4 +4,5 @@ import PropTypes from 'prop-types';
 export const defaultChildrenPropTypes = PropTypes.oneOfType([
   PropTypes.arrayOf(PropTypes.node),
   PropTypes.node,
+  PropTypes.object,
```
[the remark this answers is not in the document]
Comment (papuruth) [reply app/javascript/common-prop-types/default-children-prop-types.js:7]: I'll check on local if it isn't causing any problem then I'll push a commit

**Review by papuruth — COMMENTED** [review]
I have pushed the changes please review

**Review by aitchiss — APPROVED** [review]
Thanks for making those changes! ✨

**Review by citizen428 — APPROVED** [review]

**Pull request merged**